src/tests/system/tests/test_feature.py

```diff
@@ -0,0 +1,86 @@
+"""
+SSSD Feature Presence Tests
+
+:requirement: features
+"""
+
+from __future__ import annotations
+
+import pytest
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.topology import KnownTopology
+
+
+@pytest.mark.parametrize(
+    "distribution, distro_major, distro_minor, sssd_major, sssd_minor, feature, presence",
+    [
+        ("Fedora", 39, 0, 2, 9, "passkey", True),
+        ("CentOS Stream", 9, 0, 2, 9, "passkey", True),
+        ("Red Hat Enterprise Linux", 9, 4, 2, 9, "passkey", True),
+        ("Ubuntu", 23, 10, 2, 9, "passkey", True),
+        (None, None, None, 2, 10, "knownhosts", True),
+        ("Fedora", 40, 0, 2, 10, "files-provider", False),
+        ("CentOS Stream", 10, 0, 2, 10, "files-provider", False),
+        ("Red Hat Enterprise Linux", 10, 0, 2, 10, "files-provider", False),
+        (None, None, None, 2, 10, "ldap_use_ppolicy", True),
+        ("Fedora", 41, 0, 2, 10, "non-privileged", True),
+        ("CentOS Stream", 9, 0, 2, 10, "non-privileged", True),
+        ("Red Hat Enterprise Linux", 9, 4, 2, 10, "non-privileged", True),
```

Comment:
Why do we already put RHEL9.4::2.10::non-privileged here?
It's unclear if this will ever land RHEL9, but if it will then not earlier than 9.7
There is some sense to keep C9S::2.10::non-privileged because upstream CI runs 'master' branch on C9S, but RHEL9... why...

Reply:
That was just an assumption. It's quite difficult to follow the feature development across distributions that share development branches (c9s/rhel9). I'm fine with removing this line. Is that what you are suggesting?

Reply:
I would remove it if it's not required for obscure reasons.

Reply:
Thanks, but now we need to add RHEL10.0::2.10::non-privileged

Reply:
You added "files-provider" instead of "non-privileged".

Reply:
:facepalm: